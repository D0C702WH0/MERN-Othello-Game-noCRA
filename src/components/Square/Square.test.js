import 'jsdom-global/register'
import React from 'react'
import { configure, mount } from 'enzyme'
import TestRenderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Square from './Square'

configure({ adapter: new Adapter() })

let renderer
const click = jest.fn()
beforeEach(() => {
  renderer = mount(
    <Square
      rowI={3}
      colI={3}
      value={'WHITE'}
      click={click} />
  )
})

describe('Square', () => {
  it('Basic rendering', () => {
    const renderer = TestRenderer.create(
      <Square value={'WHITE'}/>)
    const result = renderer.toJSON()
    expect(result).toMatchSnapshot()
  })

  it('should render one pawn', () => {
    expect(renderer.find('img')).toHaveLength(1)
  })

  it('should render one pawn with className "yellow"', () => {
    expect(renderer.find('.yellow')).toHaveLength(1)
  })

  it('should render one pawn with className "red" on props value={"BLACK"}', () => {
    renderer.setProps({ value: 'BLACK' })
    expect(renderer.find('.red')).toHaveLength(1)
  })
})
