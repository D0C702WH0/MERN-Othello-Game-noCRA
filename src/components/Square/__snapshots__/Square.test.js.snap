// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square Basic rendering 1`] = `
<DocumentFragment>
  <div
    class="zoom"
    tabindex="0"
  >
    <div
      class="square"
      data-testid="square"
      tabindex="-1"
    >
      <img
        alt="duck pawn"
        class="duckFace yellow"
        data-testid="test-pawn-yellow"
        src="./assets/img/duck-yellow.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
